lib/fixed-points.nix

```
@@ -311,28 +311,40 @@ rec {
     );
 
   /**
-    Compose two extending functions of the type expected by 'extends'
-    into one where changes made in the first are available in the
-    'super' of the second
+    Compose two [`overlays`](#chap-overlays) and returns the resulting attribute set.
```

Coming back to this after writing my other comments, I think a reference to `extends` is quite relevant, and could serve the reader well.
Perhaps we could illustrate this with an example?

````suggestion
    Compose two [`overlays`](#chap-overlays) and returns the resulting attribute set.

    ```nix
    let
      original =
        final: { r = final.a + final.b; a = 1; b = 2; };
      f =
        final: prev: { b = final.c * prev.b; c = 3; };
      g =
        final: prev: { c = 10; cPrev = prev.c; r = toString prev.r; };

      # Compute the fixed point.
      # As we only call our functions once, `final` above only takes on the value of `fixedpoint`.
      fixedpoint =
        lib.fix
          (lib.extends
            (lib.composeExtensions f g)
            original
          );
    in fixedpoint
    => {
      a = 1;
      b = 20;
      c = 10;
      cPrev = 3;
      r = "21";
    }
    ```
    
    See also [`fix`](#function-library-lib.fixedPoints.fix), [`extends`](#function-library-lib.fixedPoints.extends).
````